### PR TITLE
Merge and remove duplicate README file, lint code snippets

### PR DESCRIPTION
**Pull request by runejuhl**

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -66,59 +66,60 @@ The rules in the `pre` and `post` classes are fairly general. These two classes
...
+class my_fw::pre {
+  Firewall {
+    require => undef,
+  }
+
+  # Default firewall rules
+  firewall { '000 accept all icmp':
+    proto  => 'icmp',
+    action => 'accept',
+  }->
```
Comment (bastelfreak) [inline README.markdown]:
```suggestion
  }
```

File: README.markdown [hunk 2]
```diff
@@ -66,59 +66,60 @@ The rules in the `pre` and `post` classes are fairly general. These two classes
...
+  firewall { '001 accept all to lo interface':
```
Comment (bastelfreak) [inline README.markdown]:
```suggestion
  -> firewall { '001 accept all to lo interface':
```

File: README.markdown [hunk 3]
```diff
@@ -66,59 +66,60 @@ The rules in the `pre` and `post` classes are fairly general. These two classes
...
+    proto   => 'all',
+    iniface => 'lo',
+    action  => 'accept',
+  }->
```
Comment (bastelfreak) [inline README.markdown]:
```suggestion
  }
```

File: README.markdown (hunk 2 above)
Comment (runejuhl) [reply README.markdown]: Good catch!

File: README.markdown [hunk 4]
```diff
@@ -66,59 +66,60 @@ The rules in the `pre` and `post` classes are fairly general. These two classes
...
+  firewall { '002 reject local traffic not on loopback interface':
```
Comment (bastelfreak) [inline README.markdown]:
```suggestion
  -> firewall { '002 reject local traffic not on loopback interface':
```

File: README.markdown [hunk 5]
```diff
@@ -66,59 +66,60 @@ The rules in the `pre` and `post` classes are fairly general. These two classes
...
+    iniface     => '! lo',
+    proto       => 'all',
+    destination => '127.0.0.1/8',
+    action      => 'reject',
+  }->
```
Comment (bastelfreak) [inline README.markdown]:
```suggestion
  }
```

File: README.markdown [hunk 6]
```diff
@@ -66,59 +66,60 @@ The rules in the `pre` and `post` classes are fairly general. These two classes
...
+  firewall { '003 accept related established rules':
```
Comment (bastelfreak) [inline README.markdown]:
```suggestion
  -> firewall { '003 accept related established rules':
```

**Pull request merged**